peachjam/adapters/judgments.py

```diff
@@ -0,0 +1,309 @@
+                number=case_number["number"],
+                string=case_number["string"],
+            )
+
+    def get_judges(self, judges, doc):
+        doc.judges.clear()
+        if judges:
+            for judge in judges:
+                j, _ = Judge.objects.get_or_create(name=judge)
+                doc.judges.add(j)
+
+    def get_taxonomies(self, topics, doc):
+        DocumentTopic.objects.filter(document=doc).delete()
+        if topics:
+            for topic in topics:
+                # we are not building a tree here
+                taxonomy = Taxonomy.objects.filter(name=topic).first()
+                if taxonomy:
+                    DocumentTopic.objects.create(
+                        document=doc,
+                        topic=taxonomy,
+                    )
+
+    def attach_source_file(self, doc, created_doc):
+        source_file_url = (
+            f"{self.api_url}/judgments{doc['expression_frbr_uri']}/source.file"
+        )
+        try:
+            r = self.client_get(source_file_url)
+        except requests.exceptions.HTTPError as e:
+            # ignore 404s, which means there's no source file
+            if e.response.status_code == 404:
+                return
+            raise
+
+        with NamedTemporaryFile() as f:
+            f.write(r.content)
+            f.flush()
+            mimetype = magic.from_file(f.name, mime=True)
+            ext = guess_extension(mimetype) or ""
+            filename = f"{slugify(doc['title'])[:200]}{ext}"
+
+            sf, _ = SourceFile.objects.update_or_create(
+                document=created_doc,
+                defaults={
+                    "file": File(f, filename),
+                    "mimetype": mimetype,
+                },
+            )
+            sf.ensure_file_as_pdf()
+
+    def get_content_html(self, doc):
+        try:
+            url = f"{self.api_url}/judgments{doc['expression_frbr_uri']}/.html"
+            response = self.client_get(url).text
+            return response
+        except requests.exceptions.HTTPError as e:
+            if e.response.status_code == 404:
+                return ""
+            raise e
+
+    def delete_document(self, expression_frbr_uri):
+        url = f"{self.api_url}/judgments{expression_frbr_uri}"
+
+        try:
+            self.client_get(url)
+        except requests.exceptions.HTTPError as e:
+            if e.response.status_code == 404:
+                document = Judgment.objects.filter(
+                    expression_frbr_uri=expression_frbr_uri
+                ).first()
+                if document:
+                    document.delete()
+            else:
+                raise e
+
+
+@plugins.register("ingestor-adapter")
+class JudgmentDeleteAdapter(BaseJudgmentAdapter):
+    def check_for_updates(self, last_refreshed):
+        from peachjam.tasks import get_deleted_documents
+
+        # kick of bg tasks in batches
+        judgments_count = Judgment.objects.count()
```

Comment:
shouldn't this be filtered by the court code? perhaps create a queryset once and then pass it into get_deleted_documents

Reply:
Oh, I see `get_deleted_documents` is the background task. This qs should still be filtered though, right?

Reply:
added the filter